### PR TITLE
Add endpoint JSON info

### PR DESCRIPTION
**Pull request by hjvoid**
## WHAT
Update csp.js to include a JSON object containing csp endoint info as per the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-to).

### CODE REVIEW

File: app/middleware/csp.js [hunk 1]
```diff
@@ -52,8 +52,16 @@ const connectSourceCardDetails = ['\'self\'', 'https://www.google-analytics.com'
...
+const cspEndpointJson = {
+  group: CSP_REPORTING_ENDPOINT_NAME,
+  max_age: 100000,
+  endpoints: [
+    { url: `${paths.csp.path}` }
+  ]
+}
+
+const setResponseCspReportingEndpoint = (req, res, next) => {
+  res.setHeader('Report-To', cspEndpointJson)
```
Comment (iqbalgds) [inline app/middleware/csp.js:64]: Might need to Stringify the JSON object

File: server.js [hunk 2]
```diff
@@ -189,6 +189,7 @@ function initialise () {
   if (NODE_ENV !== 'test') {
     app.use(metrics.initialise())
   }
+  app.use(setReportingEndpoints) // adds reporting-endpoints header to all responses
```
Comment (iqbalgds) [inline server.js:192]: Nice

File: test/middleware/csp.test.js [hunk 3]
```diff
@@ -177,7 +177,7 @@ describe('CSP middleware', () => {
     const csp = requireHelper('../../app/middleware/csp')
 
     const next = sinon.spy()
-    const response = { setHeader: sinon.spy() }
+    const response = { setHeader: sinon.spy(), removeHeader: sinon.spy() }
```
Comment (iqbalgds) [inline test/middleware/csp.test.js:180]: Just wondering why we need to add a spy for this function?

Comment (nlsteers) [reply test/middleware/csp.test.js:180]: the helmet upgrade, presume it now removes headers in some cases and this was required to make the tests green

**Pull request merged**